virttest/utils_test/qemu/__init__.py

```diff
@@ -698,3 +699,309 @@ def get_all_memorys(self, vm):
+            logging.info("The %s device can find in the monitor." % usb_type)
+        exist = self.check_usb_dev_guest(usb_type, session, devs)
+        if exist:
+            logging.info(" %s %s device was found in guest" % (exist, usb_type)
+        else:
+            raise exceptions.TestFail("Could not find '%s' in guest" % usb_type)
+
+    def check_usb_dev_guest(self, usb_type, session, devs):
+        """
+        check if the usb device exist in guest
+        """
+        logging.info("Verify %s device in guest." % usb_type)
+        cmd ="lsusb.py"
+        output = session.cmd_output(cmd)
+        num = 0
+        for devnum in devs:
+            for i in range(1,3):
+                key = "%s-%s" % (int(devnum[0][0])+i, devnum[1])
+                if key not in output:
+                    continue
+                else:
+                    num += 1
+        if num > 0:
+            return num
+        else:
+            return False
+
+    def create_img(self, image="usb", fmt="qcow2", size="200M"):
+
+        """
+        create image file for drive
+        """
+        self.params['images'] += " %s" % image
+        self.params["image_name_%s" % image] = "images/%s" % image
+        self.params["image_format_%s" % image] = fmt
+        self.params["image_size_%s" % image] = size
+        self.params["remove_image_%s" % image] = "yes"
+        self.params["create_image_%s" % image] = "yes"
+        env_process.process_images(env_process.preprocess_image,
+                                   self.test, self.params)
+
+    def drive_set(self, image, interface='none', fmt='qcow2',
+                  serial=None, cache=None, snapshot='off'):
+
+        """
+        create and set a drive object
+        """
+        self.create_img(image)
+        usb_params = self.params.object_params(image)
+        filename = storage.get_image_filename(usb_params,
+                                              data_dir.get_data_dir())
+        drive = qdevices.QRHDrive(image)
+        drive.set_param('if', interface)
+        drive.set_param('format', fmt)
+        drive.set_param('file', filename)
+        if serial:
+            drive.set_param('serial', serial)
+        if cache:
+            drive.set_param('cache', cache)
+        if snapshot:
+            drive.set_param('snapshot', snapshot)
+        return drive
+
+    def device_set(self, usb_name, driver, controller_type=None,
+                   drive='drive_%s', bus='usbtest.0', port=None):
+
+        """
+        create and set a device
+        """
+        dev = qdevices.QDevice(driver, aobject=usb_name)
+        dev.set_param('id', 'usb-%s' % usb_name)
+        if drive:
+            dev.set_param('drive', drive % usb_name)
+        dev.set_param('bus', bus)
+        if port:
+            dev.set_param('port', port)
+        if controller_type:
+            dev.parent_bus += ({'type': controller_type},)
+        return dev
+
+    def device_add(self, dev, driver):
+
```

Comment:
But  the usb_by_params can't return a usb-stroage device and hotpulg or unplug it.so i need to define it by myself.